lib/translator/verify.cc

```diff
@@ -0,0 +1,111 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "verify.h"
+namespace translator {
+
+StatusCode VerifyToNvme(absl::Span<const uint8_t> scsi_cmd,
+                        nvme::GenericQueueEntryCmd& nvme_cmd) {
+  scsi::Verify10Command verify_cmd{};
+  if (!ReadValue(scsi_cmd, verify_cmd)) {
+    DebugLog("Malformed Verify Command");
+    return StatusCode::kInvalidInput;
+  }
+
+  // verification length of 0 is a no-op
+  if (verify_cmd.verification_length == 0) {
+    DebugLog("Varify Command is a No-Op");
+    return StatusCode::kNoTranslation;
+  }
+
+  if (verify_cmd.control_byte.naca == 1) {
+    DebugLog("Malformed Verify Command");
+    return StatusCode::kInvalidInput;
+  }
+
+  // Support for VRPROTECT requires setting PRACT and PRCHK fields of the NVM
+  // Express command
+  uint8_t prchk = 0;
+
+  if (verify_cmd.bytchk == 0) {
```

Comment:
can we split this out to a helper function, similar to what is done in https://github.com/googleinterns/scsi2nvme/pull/96/files#diff-0a609934515d59efb283d37a11dc5e5cR15 
https://github.com/googleinterns/scsi2nvme/pull/128/files#diff-db107194d03c472d8ec2cc0152a4c41bR12

Note that the rd_protect and wr_protect fields have different translations to prinfo so we probably can't have a general helper function for all 3